### PR TITLE
[OP] Add oneDNN Softmax test.

### PR DESCRIPTION
**Pull request by jmaksymc**
WIP - mkl features input error.

### CODE REVIEW

**Comment by jmaksymc** [comment]
I've finished this test, BTW normal distribution for bf16 creates NANs, here are the results for fp32 and bf16 for 2D, 3D and 4D Tensors:
image
image

File: tensorflow/core/kernels/mkl_softmax_op_test.cc [hunk 1]
```diff
@@ -1,46 +1,197 @@
...
+
+    void Run() {
+        runDefault();
+        runMkl();
+    }
+
+    void Validate() {
+        ASSERT_EQ(default_values.dtype(), mkl_values.dtype());
+        ASSERT_EQ(default_values.shape(), mkl_values.shape());
+        test::ExpectClose(default_values, mkl_values, 1e-4);
```
Comment (marvin-Yu) [inline tensorflow/core/kernels/mkl_softmax_op_test.cc:163]: Here, can we treat different data type with different data precision? 
such as, we use `1e-7` to check the `float` data type.

**Comment by marvin-Yu** [comment]
+1, looks great to me. @changqi1 pls review this PR.

File: tensorflow/core/kernels/mkl_softmax_op_test.cc (hunk 1 above)
Comment (changqi1) [reply tensorflow/core/kernels/mkl_softmax_op_test.cc:163]: @jmaksymc

Comment (jmaksymc) [reply tensorflow/core/kernels/mkl_softmax_op_test.cc:163]: Yes, working on it.

**Comment by jmaksymc** [comment]
Changed precision for different DataTypes.
